oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/version/VersionCopier.java

```diff
@@ -111,6 +125,72 @@ public boolean copyVersionHistory(String versionableUuid, Calendar minDate, bool
         return false;
     }
 
+    private boolean hasNoConflicts(String versionHistoryPath, String versionableUuid, boolean preserveOnTarget, NodeState sourceVersionHistory) {
+        // if preserveOnTarget is true then check no conflicts which means version history has moved forward only
+        if (preserveOnTarget) {
```

Comment:
same as above, return if `preserveOnTarget` is false.

Reply:
same reasoning as above, the code is concise also will end up having multiple if blocks